### PR TITLE
DeleteClientAsync improvement

### PR DESCRIPTION
**Pull request by andrei-anfilets**

### CODE REVIEW

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs [hunk 1]
```diff
@@ -208,6 +224,19 @@ public async Task DeleteClientAsync(Guid clientId)
...
         }
 
+        public async Task DeleteMemberClientsWithThresholdAsync(Guid memberId, int inactiveHoursThreshold)
+        {
+            using (var connection = _sqlConnectionFactory.CreateConnection())
+            {
+                var sqlQuery = $@"
+                    DELETE FROM Clients 
+                    WHERE 
+                        {nameof(Client.MemberId)} = @{nameof(memberId)} AND DATEDIFF(MINUTE, {nameof(Client.LastClientActivity)}, GETUTCDATE()) > @{nameof(inactiveHoursThreshold)} * 60";
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs]: I think that if you decided to convert hours into minutes than better to replace inactiveHoursThreshold with inactiveMinutesThreshold.

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs [hunk 2]
```diff
@@ -18,6 +18,22 @@ public ClientRepository(ISqlConnectionFactory sqlConnectionFactory) : base(sqlCo
         {
         }
 
+        public async Task<bool> MemberHasClientsAsync(Guid memberId)
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs]: Please write test for this repository method.

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs [hunk 3]
```diff
@@ -208,6 +224,19 @@ public async Task DeleteClientAsync(Guid clientId)
             }
         }
 
+        public async Task DeleteMemberClientsWithThresholdAsync(Guid memberId, int inactiveHoursThreshold)
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs]: Please write test for this repository method.

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs [hunk 4]
```diff
@@ -14,6 +14,8 @@ public interface IClientRepository
         Task AddClientAsync(Client client);
         Task UpdateClientAsync(Client client);
         Task DeleteClientAsync(Guid clientId);
+        Task DeleteMemberClientsWithThresholdAsync(Guid memberId, int inactiveHoursThreshold);
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: Unobvious name from my perspective, rename to something like that DeleteOverThresholdMemberClientsAsync

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs [hunk 5]
```diff
@@ -14,6 +14,8 @@ public interface IClientRepository
...
+        Task<bool> MemberHasClientsAsync(Guid memberId);
```
Comment (eugepae) [inline src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: IsMemberHasClientsAsync maybe better

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs (hunk 4 above)
Comment (eugenekirvel) [reply src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: It's fine for me

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs (hunk 5 above)
Comment (eugenekirvel) [reply src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: If you think that method should be renamed then better to have DoesMemberHaveClientsAsync

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs (hunk 4 above)
Comment (eugepae) [reply src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: Current name don't reflex that will be deleted all clients with exceeded threshold just threshold.

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs (hunk 5 above)
Comment (andrei-anfilets) [reply src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: renamed

File: src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs (hunk 4 above)
Comment (andrei-anfilets) [reply src/Softeq.NetKit.Chat.Data.Persistent/Repositories/IClientRepository.cs]: renamed

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs (hunk 3 above)
Comment (andrei-anfilets) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs]: done

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs (hunk 2 above)
Comment (andrei-anfilets) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs]: done

File: src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs (hunk 1 above)
Comment (andrei-anfilets) [reply src/Softeq.NetKit.Chat.Data.Persistent.Sql/Repositories/ClientRepository.cs]: replaced

**Pull request merged**